kong/plugins/oidc/handler.lua

```diff
@@ -29,27 +29,52 @@ function CustomHandler:access(config)
 
     session.configure(config)
 
-    local res, err = require("resty.openidc").authenticate(oidcConfig)
+    if tryIntrospect(oidcConfig) then
 
-    if err then
-      if config.recovery_page_path then
-        ngx.log(ngx.DEBUG, "Entering recovery page: " .. config.recovery_page_path)
-        return ngx.redirect(config.recovery_page_path)
+      ngx.log(ngx.DEBUG, "In plugin CustomHandler:proceeding with two legged authentication, requested path: " .. ngx.var.request_uri)
+
+    else
+
+      local res, err = require("resty.openidc").authenticate(oidcConfig)
+
+      if err then
+        if config.recovery_page_path then
+          ngx.log(ngx.DEBUG, "Entering recovery page: " .. config.recovery_page_path)
+          return ngx.redirect(config.recovery_page_path)
+        end
+        utils.exit(500, err, ngx.HTTP_INTERNAL_SERVER_ERROR)
+      end
+
+      if res and res.user then
+        utils.injectUser(res.user)
```

Comment:
The user field isn't, when the user uses this flow. So Kong doesn't know, that the requests is already authenticated. You would need to set minimum the `authenticated_consumer.id` field from the subject claim.

Reply:
Fixed -> now providing sub field as consumer id